### ISSUE TITLE
Kudos, not problems!

### ISSUE THREAD

**Issue by fotodok** [Kemmey/Kemmey-TeslaWatch-Public#137]
This is my third attempt to use wearable software to control my Model 3. Thus far it is the least expensive and the BEST design. It provides the simplest mechanism to open the charge post which is my most used feature. My only constructive feedback is the annoyance of entering Tesla login credentials. Trying to do it from the watch is very tedious. Trying to enter the text info using the associated Apple device option has been inconsistently available in my hands.

Thanks for a great value.
Smiles and hugs!

**Comment by Kemmey** [comment]
Thank you so much! Very much appreciated! I'm glad you like it :-D 

And yeah, the login experience is abysmal. It really comes down to this project being an independent watch app, which was made possible with watchOS 6. However, Apple still have so many issues with the iOS integration - it basically doesn't work - that I really regret making it independent. It should have an iOS companion solely for login, as many watch apps do. However, once an app has been deployed to the store as independent, it can NEVER have an iOS companion app that can utilize watchconnectivity - which is just ridiculous. So for me it is an unsolvable problem. I do have a SEPARATE app that can do the login on iOS and transfer the token to the watch - but then you'd have to delete those apps after login, as that's all they do. And you still have to fight the iOS integration with watchconnectivity which is also quite broken...

**Comment by fotodok** [comment]
Wow. What a headache. And today I read of an ongoing battle w Apple over the unofficial APIs and Tesla dragging their feet. I wrote an enthusiastic appeal to Tesla today encouraging them to work with you and other developers by formally condoning use of the API.

Geez… I just found you and am excited about using the app w the lastest OS and a series 6 watch.
Now to find out about this… really too much,a s I’m sure you agree!

Fingers crossed for us both, though you’ve got a lot more on the table.  ; D
Good luck.
Mark

On Sep 17, 2020, at 4:32 PM, Kim Hansen <notifications@github.com> wrote:


Thank you so much! Very much appreciated! I'm glad you like it :-D

And yeah, the login experience is abysmal. It really comes down to this project being an independent watch app, which was made possible with watchOS 6. However, Apple still have so many issues with the iOS integration - it basically doesn't work - that I really regret making it independent. It should have an iOS companion solely for login, as many watch apps do. However, once an app has been deployed to the store as independent, it can NEVER have an iOS companion app that can utilize watchconnectivity - which is just ridiculous. So for me it is an unsolvable problem. I do have a SEPARATE app that can do the login on iOS and transfer the token to the watch - but then you'd have to delete those apps after login, as that's all they do. And you still have to fight the iOS integration with watchconnectivity which is also quite broken...

**Comment by fotodok** [comment]
Hi.
Installing my favorite app on a new watch.
Inputting correct user name and password but getting error message:
Could not acquire token: Request Failed
Response code: 401
Message: Unauthorized

Any suggestions?

Looked at the token method and it was terribly confusing…
Mark

**Comment by Kemmey** [comment]
Usually 401 means your username or password is wrong, but please note that Tesla currently has an ongoing global outage. Specifically the Sign-In servers are affected.

**Comment by Kemmey** [comment]
Please verify you can log into your account on Tesla.com and triple check your password when entering.

**Comment by fotodok** [comment]
Good call. The problem with login was on Tesla’s side. Still not working!

“Call is back tomorrow.”
Good grief

Mark
